library/src/main/java/bio/terra/landingzone/stairway/flight/create/resource/step/CreateStorageAccountDNSZoneGroupStep.java

```diff
@@ -0,0 +1,90 @@
+package bio.terra.landingzone.stairway.flight.create.resource.step;
+
+import bio.terra.landingzone.common.DnsZoneGroupResourceHelper;
+import bio.terra.landingzone.library.landingzones.definition.ArmManagers;
+import bio.terra.landingzone.library.landingzones.definition.ResourceNameGenerator;
+import bio.terra.landingzone.service.landingzone.azure.model.LandingZoneResource;
+import bio.terra.landingzone.stairway.flight.ResourceNameProvider;
+import bio.terra.landingzone.stairway.flight.ResourceNameRequirements;
+import bio.terra.stairway.FlightContext;
+import com.azure.resourcemanager.network.fluent.models.PrivateDnsZoneGroupInner;
+import com.azure.resourcemanager.network.models.PrivateDnsZoneConfig;
+import java.util.List;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Creates a DNS Zone Group, which is a link between a private endpoint and a private DNS Zone. */
+public class CreateStorageAccountDNSZoneGroupStep extends BaseResourceCreateStep {
+  private static final Logger logger =
+      LoggerFactory.getLogger(CreateStorageAccountDNSZoneGroupStep.class);
+  public static final String STORAGE_ACCOUNT_DNS_ZONE_GROUP_ID =
+      "STORAGE_ACCOUNT_DNS_ZONE_GROUP_ID";
+
+  public CreateStorageAccountDNSZoneGroupStep(
+      ArmManagers armManagers, ResourceNameProvider resourceNameProvider) {
+    super(armManagers, resourceNameProvider);
+  }
+
+  @Override
+  protected void createResource(FlightContext context, ArmManagers armManagers) {
+    var privateEndpoint =
+        getParameterOrThrow(
+            context.getWorkingMap(),
+            CreateStorageAccountPrivateEndpointStep.STORAGE_ACCOUNT_PRIVATE_ENDPOINT_RESOURCE_KEY,
+            LandingZoneResource.class);
+
+    var privateDnsId =
+        getParameterOrThrow(
+            context.getWorkingMap(),
+            CreateStorageAccountDNSZoneStep.STORAGE_ACCOUNT_DNS_ID,
+            String.class);
+
+    var dnsZoneGroup =
```

Comment:
Same question about error handling on this one, we have some base resource creation handling but it'd be good to put some thought into this particular case.